### ISSUE TITLE
DB Files on windows share drive does not open

### ISSUE THREAD

**Issue by emadmahdi** [sqlitebrowser/sqlitebrowser#2966]
It gives this error message:

![image](https://user-images.githubusercontent.com/7454511/152252325-5627a1b7-0061-4be1-b36d-27b6ef6e23de.png)

I am sure the DB file is not being used by anything

**Comment by chrisjlocke** [comment]
Is this via a drive letter which points to a network (eg, e:\some_share\directory\file.db) or a UNC path (eg, \\server\directory\file.db)

I know there is a specific 'entry' you have to use when connecting to a network database when using the system.data.dll wrapper, but don't think this is related to that.

**Comment by emadmahdi** [comment]
UNC Path like this one:
\\devicename\folder1\folder2\dbfile.db

**Comment by chrisjlocke** [comment]
For the avoidance of doubt, if you copy that db file locally, does it open OK?

**Comment by emadmahdi** [comment]
Yes, it open normally ...

**Comment by emadmahdi** [comment]
I don't know if this can make a difference ... I am not using regular windows share ... I am using android share using an app named Lan Drive .. Below are the android/apple store links for it .. it is very simple .. just install it on any android device or apple device then add the folders you want to share then start the service
.
https://play.google.com/store/apps/details?id=fr.webrox.landrive
.
https://apps.apple.com/us/app/lan-drive-samba-server-client/id1317727404
.

**Comment by chrisjlocke** [comment]
Aah right - that would make more sense.  If it's not a standard network protocol it won't be able to write the necessary journal files, thus it thinks the database is locked.  
So the workaround is to copy it locally, make the changes then copy the file back again.

**Comment by emadmahdi** [comment]
I'm already doing this trick ... but i was wondering if you can fix this issue ?!!
the Lan Dirve software is just providing samba share and it is kind of standard and used widely for more than 20 years !!

**Comment by emadmahdi** [comment]
samaba as prtocol was first implmnted to allow microsoft windows to access unix systems ... but now it is also supporting almost all operating systems

**Comment by rwibm** [comment]
Is this issue still being looked into?

It is still happening on Windows with a standard SMB network share/mapped drive. I've tried both mapped drive and \\ip\share notation, doesn't work on either and comes up with the same message saying the file is locked (when it isnt).

Python has no problems loading this file from a mapped drive and accessing it so I'm pretty sure this issue is db browser-specific as it works absolutely fine if I copy the file to my local machine and access it. But constantly copying back and forth kinda defeats the point of using the GUI, I may as well just make updates via ssh at that point.

I did try looking into this and the only thing I noticed is that it shows the same error even when opening in read-only mode, I'd have assumed read-only mode wouldn't care about a lock? So perhaps something is failing when checking the lock status over SMB and the app just thinks its locked because it failed to check or something?

**Comment by chrisjlocke** [comment]
What sort of network share are you connecting to .. ie, is it a Linux based NAS, or a Windows server, etc.
What version of DB4S are you using?
How are you opening the file?  I created a 'test.db' file on my Synology box (so Linux based), double-clicked on it in Windows and it opened fine.  Created a table with no problems.
Technically SQLite doesn't work across networks so this isn't a DB4S issue per se.  However, I've used it in a Windows environment just fine.
The obvious question is anti-virus software related.  Have you a series of anti-virus suites installed?
Usually opening a network file upsets anti-virus software.

**Comment by rwibm** [comment]
I've tried both a windows server and a linux box, same problem. Tried opening the file normally and as read-only within the app, and also double-clicking, same error comes up.

I don't see a journal file either, but I am starting to suspect its an issue with my local machine as I've tried a plugin for PyCharm that also gives the same error. I am using a corporate device so maybe they have some weird group policy causing issues.

Oh well, looks like it's a me problem. Luckily I don't need to use this very often so I'll have to go without it.

Thanks for all your help anyway :)

**Comment by rwibm** [comment]
Yeah it's not permissions unfortunately, the user has full permissions on the folder, its used as a remote dev drive and nothing else has any issues editing, removing, or adding new files, just seems to be that lock that's the problem. It would be nice if opening the file in "read-only" mode just ignored the lock instead of failing, I'm not even sure why it tries to acquire a lock in read-only mode 🤷  The db opens up fine in things like notepad, python, and other non-db apps so its definitely something to do with the lock.

I think I figured out what's wrong anyway; it turned out to be 2 separate issues that showed up as the same problem.

I've included details below in case it helps anybody else in the future:

*Windows Server SMB*
This seems to be some weird group policy preventing the files being accessed properly, I'm not worried about this, nor do I know exactly what is preventing it from working.

*Linux SMB*
This is the one I have been trying to get working for a while and I've figured out what the problem is: WSL (windows subsystem for linux)

If I use the normal smbd within ubuntu I can access the file absolutely fine, locks and all. But WSL2 changes the IP address on every boot, which was breaking my mapped drives, so I started using the built-in \\wsl$\ file share which uses 9P internally (Plan 9 File Protocol) rather than SMB.

I completely forgot I swapped my mapped drive to 9P instead of SMB and I'm guessing this is where the problem is coming from, maybe it doesn't support locking files? This is the first time I've ran into a problem like this so I've got no clue. For now I'll just use the standard SMB server and will have to look up the IP whenever I need to access it.

Thanks for your help!

**Comment by justinclift** [comment]
Interesting.  Well, at least you have a workaround. :smile:

**Comment by danielsjf** [comment]
Came here from the WSL bug page. Can confirm that network files on my Synology can be opened and therefore this is not really a duplicate of the WSL error.
